pkg/calculator/experimentcalc/experiment_calculator.go

```diff
@@ -0,0 +1,613 @@
+		}
+		cvr := 0.0
+		if srcVrs[i].EvaluationCount.UserCount != 0 {
+			cvr = float64(srcVrs[i].ExperimentCount.UserCount) / float64(srcVrs[i].EvaluationCount.UserCount)
+		}
+		goalResult.VariationResults[i].CvrTimeseries = &eventcounter.Timeseries{
+			Timestamps: []int64{timestamp},
+			Values:     []float64{cvr},
+		}
+		valuePerUser := 0.0
+		if srcVrs[i].ExperimentCount.UserCount != 0 {
+			valuePerUser = srcVrs[i].ExperimentCount.ValueSum / float64(srcVrs[i].ExperimentCount.UserCount)
+		}
+		goalResult.VariationResults[i].GoalValueSumPerUserTimeseries = &eventcounter.Timeseries{
+			Timestamps: []int64{timestamp},
+			Values:     []float64{valuePerUser},
+		}
+		goalResult.VariationResults[i].GoalValueSumPerUserMedianTimeseries = &eventcounter.Timeseries{
+			Timestamps: []int64{timestamp},
+			Values:     []float64{srcVrs[i].GoalValueSumPerUserProb.Median},
+		}
+		goalResult.VariationResults[i].GoalValueSumPerUserPercentile025Timeseries = &eventcounter.Timeseries{
+			Timestamps: []int64{timestamp},
+			Values:     []float64{srcVrs[i].GoalValueSumPerUserProb.Percentile025},
+		}
+		goalResult.VariationResults[i].GoalValueSumPerUserPercentile975Timeseries = &eventcounter.Timeseries{
+			Timestamps: []int64{timestamp},
+			Values:     []float64{srcVrs[i].GoalValueSumPerUserProb.Percentile975},
+		}
+	}
+}
+
+func (e ExperimentCalculator) binomialModelSample(
+	ctx context.Context,
+	vids []string,
+	goalUc, evalUc []int64,
+	baseLineIdx int,
+) map[string]*eventcounter.VariationResult {
+	//The index starts from 1 in PyStan.
+	baseLineIdx++
+	samplesChan := make(chan dataframe.DataFrame)
+	wg := sync.WaitGroup{}
+	wg.Add(numOfChains)
+	for i := 1; i <= numOfChains; i++ {
+		go func(chain int) {
+			defer wg.Done()
+			compiledModel, err := e.stan.CompileModel(ctx, stan.ModelCode())
+			if err != nil {
+				e.logger.Error("failed to compile model",
+					log.FieldsFromImcomingContext(ctx).AddFields(
+						zap.Error(err),
+					)...,
+				)
+				return
+			}
+			modelID := compiledModel.Name[len("models/"):]
+			req := stan.CreateFitReq{
+				Chain: chain,
+				Data: map[string]interface{}{
+					"g": len(goalUc),
+					"x": goalUc,
+					"n": evalUc,
+				},
+				Function:   stan.HmcNUTSFunction,
+				NumSamples: 21000,
+				NumWarmup:  1000,
+				RandomSeed: 1234,
+			}
+			fitResp, err := e.stan.CreateFit(ctx, modelID, req)
+			if err != nil {
+				e.logger.Error("failed to create fit",
+					log.FieldsFromImcomingContext(ctx).AddFields(
+						zap.String("modelId", modelID),
+						zap.Error(err),
+					)...,
+				)
+			}
+			fitId := fitResp[len("operations/"):]
+			for {
+				details, err := e.stan.GetOperationDetails(ctx, fitId)
+				if err != nil {
+					e.logger.Error("failed to get operation details",
+						log.FieldsFromImcomingContext(ctx).AddFields(
+							zap.String("fitId", fitId),
+							zap.Error(err),
+						)...,
+					)
+				}
+				if details.Done {
+					break
+				}
+				time.Sleep(50 * time.Millisecond)
```

Comment:
I found that `pystan` was [using `50ms` to check result from `httpstan`](https://github.com/stan-dev/pystan/blob/main/stan/model.py#L222).
Actually when I am testing, this will return in around 1 seconds.